test/magic_test.rb

```diff
@@ -25,4 +66,11 @@ class Marcel::MimeType::MagicTest < Marcel::TestCase
     assert Marcel::Magic.child?('text/csv', 'text/plain')
     refute Marcel::Magic.child?('text/plain', 'text/csv')
   end
+
+  test "child? with aliases" do
+    Marcel::Magic.add('canonical/parent', aliases: 'alias/parent')
+    Marcel::Magic.add('canonical/child', aliases: 'alias/child', parents: 'canonical/parent')
+
+    assert Marcel::Magic.child?('alias/child', 'alias/parent')
+  end
```

Comment:
Arguments may be alias types, so check that they're resolved to canonicals before checking parentage.